from PyQt6 import QtCore, QtGui, QtWidgets
from PyQt6.QtGui import QIcon, QFont


class Ui_Calendar(object):
    def setupUi(self, MainWindow):
        MainWindow.setObjectName("MainWindow")
        MainWindow.resize(821, 600)
        self.centralwidget = QtWidgets.QWidget(parent=MainWindow)
        self.centralwidget.setObjectName("centralwidget")
        self.heading_label = QtWidgets.QLabel(parent=self.centralwidget)
        self.heading_label.setGeometry(QtCore.QRect(300, 10, 241, 31))
        font = QtGui.QFont()
        font.setFamily("Rockwell Extra Bold")
        font.setPointSize(9)
        font.setBold(True)
        font.setWeight(75)
        self.heading_label.setFont(font)
        self.heading_label.setAlignment(QtCore.Qt.AlignmentFlag.AlignCenter)
        self.heading_label.setObjectName("heading_label")
        self.second_label = QtWidgets.QLabel(parent=self.centralwidget)
        self.second_label.setGeometry(QtCore.QRect(280, 40, 301, 16))
        font = QtGui.QFont()
        font.setFamily("Rockwell Extra Bold")
        font.setPointSize(6)
        font.setBold(True)
        font.setWeight(75)
        self.second_label.setFont(font)
        self.second_label.setObjectName("second_label")
        self.data_label = QtWidgets.QLabel(parent=self.centralwidget)
        self.data_label.setGeometry(QtCore.QRect(20, 80, 401, 20))
        font = QtGui.QFont()
        font.setFamily("Rockwell Extra Bold")
        font.setBold(True)
        font.setWeight(75)
        self.data_label.setFont(font)
        self.data_label.setAlignment(QtCore.Qt.AlignmentFlag.AlignCenter)
        self.data_label.setObjectName("data_label")
        self.event_label = QtWidgets.QLabel(parent=self.centralwidget)
        self.event_label.setGeometry(QtCore.QRect(420, 80, 371, 20))
        font = QtGui.QFont()
        font.setFamily("Rockwell Extra Bold")
        font.setBold(True)
        font.setWeight(75)
        self.event_label.setFont(font)
        self.event_label.setAlignment(QtCore.Qt.AlignmentFlag.AlignCenter)
        self.event_label.setObjectName("event_label")
        self.return_button = QtWidgets.QPushButton(parent=self.centralwidget)
        self.return_button.setGeometry(QtCore.QRect(330, 520, 181, 28))
        font = QtGui.QFont()
        font.setFamily("Rockwell Extra Bold")
        font.setBold(True)
        font.setWeight(75)
        self.return_button.setFont(font)
        self.return_button.setObjectName("return_button")
        self.widget = QtWidgets.QWidget(parent=self.centralwidget)
        self.widget.setGeometry(QtCore.QRect(20, 120, 771, 391))
        self.widget.setObjectName("widget")
        self.horizontalLayout = QtWidgets.QHBoxLayout(self.widget)
        self.horizontalLayout.setContentsMargins(0, 0, 0, 0)
        self.horizontalLayout.setObjectName("horizontalLayout")
        self.calendar = QtWidgets.QCalendarWidget(parent=self.widget)
        self.calendar.setObjectName("calendar")
        self.horizontalLayout.addWidget(self.calendar)
        self.result = QtWidgets.QPlainTextEdit(parent=self.widget)
        self.result.setObjectName("result")
        self.horizontalLayout.addWidget(self.result)
        MainWindow.setCentralWidget(self.centralwidget)
        self.menubar = QtWidgets.QMenuBar(parent=MainWindow)
        self.menubar.setGeometry(QtCore.QRect(0, 0, 821, 26))
        self.menubar.setObjectName("menubar")
        MainWindow.setMenuBar(self.menubar)
        self.statusbar = QtWidgets.QStatusBar(parent=MainWindow)
        self.statusbar.setObjectName("statusbar")
        MainWindow.setStatusBar(self.statusbar)

        self.small_font = QFont('Rockwell Extra Bold', 6)
        self.average_font = QFont('Rockwell Extra Bold', 7)
        self.big_font = QFont('Rockwell Extra Bold', 9)
        self.small_font.setBold(True)
        self.average_font.setBold(True)
        self.big_font.setBold(True)

        self.heading_label.setFont(self.big_font)
        self.second_label.setFont(self.small_font)
        self.data_label.setFont(self.average_font)
        self.event_label.setFont(self.average_font)
        self.return_button.setFont(self.average_font)

        self.second_label.move(315, 40)
        self.result.setEnabled(False)
        self.return_button.setIcon(QIcon('icons/return.svg'))

        self.retranslateUi(MainWindow)
        QtCore.QMetaObject.connectSlotsByName(MainWindow)

    def retranslateUi(self, MainWindow):
        _translate = QtCore.QCoreApplication.translate
        MainWindow.setWindowTitle(_translate("MainWindow", "MainWindow"))
        self.heading_label.setText(_translate("MainWindow", "Интерактивный календарь"))
        self.second_label.setText(_translate("MainWindow", "*в календаре присутствуют только значимые даты"))
        self.data_label.setText(_translate("MainWindow", "Выберите дату для просмотра событий"))
        self.event_label.setText(_translate("MainWindow", "События, найденные по указанной дате"))
        self.return_button.setText(_translate("MainWindow", "Вернуться назад"))
